============
User guide
============


Quick start
-----------

ACEPipe is installed in ACE HPC clusters. To activate ACEPipe::

    $ source activate acepipe

To get the help or available tools in ACEPipe::

    $ acepipe -h
    $ acepipe --help

Which will return the available commands available:

.. code-block:: console

    Usage: acepipe [OPTIONS] COMMAND [ARGS]...

    Options:
      -h, --help  Show this message and exit.

    Commands:
      makedb
      runpipe
      runqc
      runsingle

To get further options available in the command::

    $ acepipe runpipe -h
    $ acepipe runpipe --help

Which will return all options available for the command:

.. code-block:: console

    Usage: acepipe runpipe [OPTIONS]

    Options:
      -h, --help                      Show this message and exit.
      -s, --sheetfile TEXT            provide sheetfile location
                                      default=samplesheet.tab
      -o, --outdir TEXT               provide output folder location
                                      default=acepipe_results
      -e, --email TEXT                provide email id
      -dt, --taxonomydb TEXT          provide reference taxonomy database location
                                      example:
                                      /usr/local/qiime_db/silva_taxonomy_db.qza
                                      default=silvadb & unitedb
      -dg, --genomedb TEXT            provide reference sequence database location
                                      example: /usr/local/qiime_db/silva_seq_db.qz
                                      default=silvadb & unitedb
      -qm, --qiimemode [forward|both]
                                      provide qiime reads mode
                                      default=forward
      -l, --readlength INTEGER        provide reads length to keep after trimming
                                      default=250
      -q, --quality INTEGER           provide quality score threshold value
                                      default=15
      -j, --joblist TEXT              provide job id as a comma-separated list
                                      default=all


Input
-----

Input file for ACEPipe is the samplesheet file which is generated by ACE LIMS and provided by the ACE Sequencing Team.
Samplesheet is a tab delimited file which contains some information such as:
    * Sample_ID
    * Sample_Name
    * Sample_Plate
    * Sample_Well
    * I7_Index_ID
    * index
    * I5_Index_ID
    * index2
    * Sample_Project
    * Description
    * Read 1 File
    * Read 2 File
    * Fwd_Primer_Name
    * Fwd_Primer_TS_seq
    * Rev_Primer_Name
    * Rev_Primer_TS_seq
    * Client_Sample_Name

ACEPipe expects the filename should shart with sample id. If your file name starts with job id, please rename the files.
The fastq file name format is ``CasavaOneEightSingleLanePerSampleDirFmt`` which naming is as ``'.+_.+_L[0-9][0-9][0-9]_R[12]_001\.fastq\.gz'``.
So the name of your files should be as: ``SAMPLEID_JOBID_OTHERID_L001_R1_001.fastq.gz``.
Example: ``SB2658_J1235_S352_L001_R1_001.fastq.gz`` and ``SB2658_J1235_S352_L001_R2_001.fastq.gz``


Output
------

ACEPipe wraps all the files with a html file. Transfer the ``result`` folder to your local computer and
open the ``index.html`` file in any internet browser (Google Chrome preferred).


Examples
--------

To activate ACEPipe::

    $ source activate acepipe

Example-1: Forward reads processing in QIIME2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To process only the forwards reads in  QIIME2::

    $ nohup acepipe runpipe -s samplesheet.tab -o output_folder -e id@eaxmple.com > acepipe_forward_output.log &

Example-2: Both reads processing in QIIME2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To process both reads in  QIIME2::

    $ nohup acepipe runpipe -s samplesheet.tab -o output_folder -qm both -e id@eaxmple.com > acepipe_both_output.log &

Example-3: Customized database
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To create customized database::

    $ acepipe makedb -dg location/to/reference/sequences/file -dt location/to/taxonomy/file -o output/to/custom/database

To use the customized database::

    $ nohup acepipe runpipe -s samplesheet.tab -dg location/to/reference/sequences/file -dt location/to/taxonomy/file
    -o output_folder > acepipe_forward_customDB_output.log &

Example-4: Customized quality control
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To process with q value of 25 and reads length of 220::

    $ nohup acepipe runpipe -s samplesheet.tab -o output_folder -q 25 -l 220 -e id@eaxmple.com > acepipe_forward_customQC_output.log &

Example-5: Single or multiple jobs processing
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

ACEPipe processes all jobs by default. However, a single job or multipe jobs can be processed.
To process a single job::

    $ nohup acepipe runpipe -s samplesheet.tab -o output_folder -J JOBID -e id@eaxmple.com > acepipe_forward_JOBID_output.log &

To process multiple jobs::

    $ nohup acepipe runpipe -s samplesheet.tab -o output_folder -J JOBID1,JOBID2,JOBID3,JOBIDX -e id@eaxmple.com > acepipe_forward_JOBS_output.log &

Example-6: Single file
~~~~~~~~~~~~~~~~~~~~~~

To use only single reads::

    $ nohup acepipe runsingle -s samplesheet.tab -o output_folder -rd forward > acepipe_single_forward_output.log &


.. note::
    Please provide your valuable comments and suggestions to make ACEPipe better.
    If you need further help, please contact Shihab Hasan (shihab.hasan@uqconnect.edu.au).